guides/common/modules/proc_host_registration_steps.adoc

```diff
@@ -0,0 +1,44 @@
+.Procedure
+
+. Generate registration command
++
+UI: In the {ProjectWebUI}, navigate to *Hosts > Register Host* and generate the registration command.
++
+Or you can generate command via API:
+ifdef::foreman-el,foreman-deb[]
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer host-registration generate-command
+
+# or manually:
+
+# curl -X POST https://{foreman-example-com}/api/registration_commands \
+    --user username:password \
+    -H 'Content-Type: application/json'
```

Comment:
Observation: This leaves your Satellite username and password in shell history. That could have pretty interesting security consequences.

It looks like we don't have this anywhere else in the documentation, so my suggestion would be to remove `:password` and according to the manual page: _If you simply specify the user name, curl will prompt for a password._

Alternatively, keep it this way but drop a sentence below explaining this threat. I personally like the opposite - having the main copy-paste bit secure by default and instructing user below about `username:password` possibility with all the consequences.

Final solution is to unset shell history for the session, you can do this with `unset HISTFILE`. This makes it a two liner, there is a trick to do this on a single line:

    HISTFILE=/dev/null curl -X POST https://{foreman-example-com}/api/registration_commands ...

Reply:
Correction: `HISTFILE=/dev/null` on a single line does not work for me, so perhaps just use `unset HISTFILE` on another line.

Reply:
it may not be such an issue if we also mention the personal access token can be used as a password and that is can have expiration set. It seems we don't have a documentation for PAT yet, otherwise I'd say just link it from here.

Reply:
When I will finish documentation for API, with examples & with described all different authentication methods we can link it here, as @ares says.

Reply:
Can you just make the option without password the default one and ask the user in the text to change "username" to "username:passwordortoken" explaining the threat? Then we can merge I think.

Reply:
password removed

Reply:
I still see it but I don't insist, let's merge.